stdlib/asm/math/u64.masm

```diff
@@ -587,12 +587,15 @@ export.rotr
     pow2
     dup
     movup.3
-    u32overflowing_mul
+    mul
+    u32split
 
     # Shift the high limb left by 32-b.
     movup.3
     movup.3
-    u32overflowing_madd
+    mul
+    add
+    u32split
```

Comment:
@bobbinth I chose not to use `u32lt` and `u32wrapping_sub` to avoid some redundant operations (and thus reduce the impact on cycle count). Do we want to use those anyway despite that? I don't have a problem with that approach, just wasn't sure what our preference was here.

Reply:
If it increases the cycle count, then let's keep it as is - but I thought that the cycle count should stay the same. Specifically:

- `u32lt` takes 3 cycles and would replace 3 operations.
- `u32wrapping_sub` takes 2 cycles and would replace 2 operations.

But maybe I missed something?

Reply:
Turns out I was thinking of something else, disregard!